src/tools/rbd_mirror/ImageReplayer.cc

```diff
@@ -176,6 +176,225 @@ void ImageReplayer<I>::BootstrapProgressContext::update_progress(
 }
 
 template <typename I>
+void ImageReplayer<I>::ReplayStatusMaker::send(Context *on_finish) {
+  dout(20) << dendl;
+
+  if (m_on_finish) {
+    dout(20) << "previous request is still in progress, ignoring" << dendl;
+    on_finish->complete(-EAGAIN);
+    return;
+  }
+
+  m_on_finish = on_finish;
+  get_registered_clients();
+}
+
+template <typename I>
+void ImageReplayer<I>::ReplayStatusMaker::get_registered_clients() {
```

Comment:
This was I tried first (the cached clients).  But I got strange results, I don't recall exactly, it looked like data were outdated or master position was behind mirror, so entries_behind_master could be negative. I did not investigate that much though (anyway I needed to query osd for other data). If you think it is unexpected and worth further investigation I will do.